### PR TITLE
Allow user to overload picking mechanism

### PR DESCRIPTION
**Pull request by deltakosh**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

File: packages/dev/core/src/Culling/ray.core.ts [hunk 1]
```diff
@@ -24,6 +24,25 @@ export type MeshPredicate = (mesh: AbstractMesh, thinInstanceIndex: number) => b
  */
 export type TrianglePickingPredicate = (p0: Vector3, p1: Vector3, p2: Vector3, ray: Ray, i0: number, i1: number, i2: number) => boolean;
 
+/**
+ * This class allows user to customize internal picking mechanism
+ */
+export class PickingCustomization {
```
Comment (RaananW) [inline packages/dev/core/src/Culling/ray.core.ts:30]: Can be an object (const), no need for the overhead of a class with a single static member

**Review by RaananW — APPROVED** [review]
Sorry, meant to approve. I would replace the class, but otherwise lgtm 🙂

File: packages/dev/core/src/Culling/ray.core.ts (hunk 1 above)
Comment (deltakosh) [reply packages/dev/core/src/Culling/ray.core.ts:30]: yes but I want to be ready for the future as we may add more customizations

Comment (RaananW) [reply packages/dev/core/src/Culling/ray.core.ts:30]: You can add more to an interface, and just create an object implementing it

Comment (deltakosh) [reply packages/dev/core/src/Culling/ray.core.ts:30]: but somehow it is not unique
What is the point then?

Comment (deltakosh) [reply packages/dev/core/src/Culling/ray.core.ts:30]: Put differently, why bothering about 2 things (interface + implementation) when a class works?

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15679/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15679/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15679/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15679/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15679/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15679/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15679/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15679/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15679/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15679/merge/testResults/webgl2playwright/index.html

File: packages/dev/core/src/Culling/ray.core.ts (hunk 1 above)
Comment (deltakosh) [reply packages/dev/core/src/Culling/ray.core.ts]: Answering my own question: There is a slight overload with classes because of prototype

**Pull request merged**